src/test/java/net/openhft/chronicle/wire/marshallable/CNFREOnMissingClassTest.java

```diff
@@ -35,4 +35,23 @@ public void throwClassNotFoundRuntimeExceptionOnMissingClassForField() {
         Jvm.startup().on(CNFREOnMissingClassTest.class, "Value of " + key + ": " + Jvm.getBoolean(key));
         final TwoFields simple = Marshallable.fromString(simpleObject);
     }
+
+    static class UsesTwoFields extends AbstractMarshallableCfg {
+        private TwoFields bothFields;
+        private String name;
+    }
+
+    /**
+     * Failing to load a class for a field with a type of java.lang.Object causes the correct behaviour but in
+     * an unexpected code path (the check for a classloader at TextWire#typeOrPrefixObject - line 1913
+     */
+    @Test(expected = ClassNotFoundRuntimeException.class)
+    public void throwClassNotFoundRuntimeExceptionOnMissingClassForFieldNotObject() {
```

Comment:
@peter-lawrey this is the failing test, and this is the customer use case

Reply:
Fixed